sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2100,13 +2103,17 @@ public TableReference getOrCreateTable(BigQueryOptions options, String tableSpec
       return tableReference;
     }
 
-    /** Writes the accumulated rows into BigQuery with streaming API. */
-    private void flushRows(Bigquery client, TableReference tableReference,
-        List<TableRow> tableRows, List<String> uniqueIds, BigQueryOptions options) {
+    /**
+     * Writes the accumulated rows into BigQuery with streaming API.
+     */
+    private void flushRows(TableReference tableReference,
+        List<TableRow> tableRows, List<String> uniqueIds, BigQueryOptions options)
+            throws InterruptedException {
       if (!tableRows.isEmpty()) {
         try {
-          BigQueryTableInserter inserter = new BigQueryTableInserter(client, options);
-          inserter.insertAll(tableReference, tableRows, uniqueIds, byteCountAggregator);
+          long totalBytes = bqServices.getDatasetService(options).insertAll(
+              tableReference, tableRows, uniqueIds);
+          byteCountAggregator.addValue(totalBytes);
```

Comment:
If you move `BigQueryTableInserter` in the BigqueryServices, how can I insert the whole `List<TableRow>` into a Bigquery table? I could use `com.google.api.services.bigquery.Bigquery` service rather then the old BigQueryTableInserte, but I can't access `getDatasetService(options)` from that object so I can't use insertAll() method, so how can I replace the following code now?
```
BigQueryTableInserter bigQueryTableInserter = new BigQueryTableInserter(bigqueryService);
   bigQueryTableInserter.getOrCreateTable(tableReference,
   BigQueryIO.Write.WriteDisposition.WRITE_APPEND,
   BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED,
   tableSchema);
bigQueryTableInserter.insertAll(tableReference, rows);
```

I am struggling with it.

Thank you